// react-native-leaflet-maps/lib/commonjs/Components/types.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.OWN_POSTION_MARKER_ID = exports.MapShapeType = exports.MapLayerType = exports.LeafletWebViewEvents = exports.INFINITE_ANIMATION_ITERATIONS = exports.AnimationType = exports.AnimationDirection = void 0;
const OWN_POSTION_MARKER_ID = 'OWN_POSTION_MARKER_ID';
exports.OWN_POSTION_MARKER_ID = OWN_POSTION_MARKER_ID;
let LeafletWebViewEvents; //  ON_MAP_SHAPE_CLICKED = "onMapShapeClicked" cannot click on shapes yet
exports.LeafletWebViewEvents = LeafletWebViewEvents;
(function (LeafletWebViewEvents) {
  LeafletWebViewEvents["MAP_COMPONENT_MOUNTED"] = "MAP_COMPONENT_MOUNTED";
  LeafletWebViewEvents["MAP_READY"] = "MAP_READY";
  LeafletWebViewEvents["DOCUMENT_EVENT_LISTENER_ADDED"] = "DOCUMENT_EVENT_LISTENER_ADDED";
  LeafletWebViewEvents["WINDOW_EVENT_LISTENER_ADDED"] = "WINDOW_EVENT_LISTENER_ADDED";
  LeafletWebViewEvents["UNABLE_TO_ADD_EVENT_LISTENER"] = "UNABLE_TO_ADD_EVENT_LISTENER";
  LeafletWebViewEvents["DOCUMENT_EVENT_LISTENER_REMOVED"] = "DOCUMENT_EVENT_LISTENER_REMOVED";
  LeafletWebViewEvents["WINDOW_EVENT_LISTENER_REMOVED"] = "WINDOW_EVENT_LISTENER_REMOVED";
  LeafletWebViewEvents["ON_MOVE_END"] = "onMoveEnd";
  LeafletWebViewEvents["ON_MOVE_START"] = "onMoveStart";
  LeafletWebViewEvents["ON_MOVE"] = "onMove";
  LeafletWebViewEvents["ON_RESIZE"] = "onResize";
  LeafletWebViewEvents["ON_UNLOAD"] = "onUnload";
  LeafletWebViewEvents["ON_VIEW_RESET"] = "onViewReset";
  LeafletWebViewEvents["ON_ZOOM_END"] = "onZoomEnd";
  LeafletWebViewEvents["ON_ZOOM_LEVELS_CHANGE"] = "onZoomLevelsChange";
  LeafletWebViewEvents["ON_ZOOM_START"] = "onZoomStart";
  LeafletWebViewEvents["ON_ZOOM"] = "onZoom";
  LeafletWebViewEvents["ON_MAP_TOUCHED"] = "onMapClicked";
  LeafletWebViewEvents["ON_MAP_MARKER_CLICKED"] = "onMapMarkerClicked";
})(LeafletWebViewEvents || (exports.LeafletWebViewEvents = LeafletWebViewEvents = {}));
let AnimationType;
exports.AnimationType = AnimationType;
(function (AnimationType) {
  AnimationType["BOUNCE"] = "bounce";
  AnimationType["FADE"] = "fade";
  AnimationType["PULSE"] = "pulse";
  AnimationType["JUMP"] = "jump";
  AnimationType["SPIN"] = "spin";
  AnimationType["WAGGLE"] = "waggle";
})(AnimationType || (exports.AnimationType = AnimationType = {}));
let MapLayerType;
exports.MapLayerType = MapLayerType;
(function (MapLayerType) {
  MapLayerType["IMAGE_LAYER"] = "ImageOverlay";
  MapLayerType["TILE_LAYER"] = "TileLayer";
  MapLayerType["VECTOR_LAYER"] = "VectorLayer";
  MapLayerType["VIDEO_LAYER"] = "VideoOverlay";
  MapLayerType["WMS_TILE_LAYER"] = "WMSTileLayer";
})(MapLayerType || (exports.MapLayerType = MapLayerType = {}));
let MapShapeType;
exports.MapShapeType = MapShapeType;
(function (MapShapeType) {
  MapShapeType["CIRCLE"] = "Circle";
  MapShapeType["CIRCLE_MARKER"] = "CircleMarker";
  MapShapeType["POLYLINE"] = "Polyline";
  MapShapeType["POLYGON"] = "Polygon";
  MapShapeType["RECTANGLE"] = "Rectangle";
})(MapShapeType || (exports.MapShapeType = MapShapeType = {}));
const INFINITE_ANIMATION_ITERATIONS = 'infinite';
exports.INFINITE_ANIMATION_ITERATIONS = INFINITE_ANIMATION_ITERATIONS;
let AnimationDirection;
exports.AnimationDirection = AnimationDirection;
(function (AnimationDirection) {
  AnimationDirection["NORMAL"] = "nomal";
  AnimationDirection["REVERSE"] = "reverse";
  AnimationDirection["ALTERNATE"] = "alternate";
  AnimationDirection["ALTERNATE_REVERSE"] = "alternate-reverse";
})(AnimationDirection || (exports.AnimationDirection = AnimationDirection = {}));
