// react-native-leaflet-maps/lib/module/Components/types.js
export const OWN_POSTION_MARKER_ID = 'OWN_POSTION_MARKER_ID';
export let LeafletWebViewEvents; //  ON_MAP_SHAPE_CLICKED = "onMapShapeClicked" cannot click on shapes yet
(function (LeafletWebViewEvents) {
  LeafletWebViewEvents["MAP_COMPONENT_MOUNTED"] = "MAP_COMPONENT_MOUNTED";
  LeafletWebViewEvents["MAP_READY"] = "MAP_READY";
  LeafletWebViewEvents["DOCUMENT_EVENT_LISTENER_ADDED"] = "DOCUMENT_EVENT_LISTENER_ADDED";
  LeafletWebViewEvents["WINDOW_EVENT_LISTENER_ADDED"] = "WINDOW_EVENT_LISTENER_ADDED";
  LeafletWebViewEvents["UNABLE_TO_ADD_EVENT_LISTENER"] = "UNABLE_TO_ADD_EVENT_LISTENER";
  LeafletWebViewEvents["DOCUMENT_EVENT_LISTENER_REMOVED"] = "DOCUMENT_EVENT_LISTENER_REMOVED";
  LeafletWebViewEvents["WINDOW_EVENT_LISTENER_REMOVED"] = "WINDOW_EVENT_LISTENER_REMOVED";
  LeafletWebViewEvents["ON_MOVE_END"] = "onMoveEnd";
  LeafletWebViewEvents["ON_MOVE_START"] = "onMoveStart";
  LeafletWebViewEvents["ON_MOVE"] = "onMove";
  LeafletWebViewEvents["ON_RESIZE"] = "onResize";
  LeafletWebViewEvents["ON_UNLOAD"] = "onUnload";
  LeafletWebViewEvents["ON_VIEW_RESET"] = "onViewReset";
  LeafletWebViewEvents["ON_ZOOM_END"] = "onZoomEnd";
  LeafletWebViewEvents["ON_ZOOM_LEVELS_CHANGE"] = "onZoomLevelsChange";
  LeafletWebViewEvents["ON_ZOOM_START"] = "onZoomStart";
  LeafletWebViewEvents["ON_ZOOM"] = "onZoom";
  LeafletWebViewEvents["ON_MAP_TOUCHED"] = "onMapClicked";
  LeafletWebViewEvents["ON_MAP_MARKER_CLICKED"] = "onMapMarkerClicked";
})(LeafletWebViewEvents || (LeafletWebViewEvents = {}));
export let AnimationType;
(function (AnimationType) {
  AnimationType["BOUNCE"] = "bounce";
  AnimationType["FADE"] = "fade";
  AnimationType["PULSE"] = "pulse";
  AnimationType["JUMP"] = "jump";
  AnimationType["SPIN"] = "spin";
  AnimationType["WAGGLE"] = "waggle";
})(AnimationType || (AnimationType = {}));
export let MapLayerType;
(function (MapLayerType) {
  MapLayerType["IMAGE_LAYER"] = "ImageOverlay";
  MapLayerType["TILE_LAYER"] = "TileLayer";
  MapLayerType["VECTOR_LAYER"] = "VectorLayer";
  MapLayerType["VIDEO_LAYER"] = "VideoOverlay";
  MapLayerType["WMS_TILE_LAYER"] = "WMSTileLayer";
})(MapLayerType || (MapLayerType = {}));
export let MapShapeType;
(function (MapShapeType) {
  MapShapeType["CIRCLE"] = "Circle";
  MapShapeType["CIRCLE_MARKER"] = "CircleMarker";
  MapShapeType["POLYLINE"] = "Polyline";
  MapShapeType["POLYGON"] = "Polygon";
  MapShapeType["RECTANGLE"] = "Rectangle";
})(MapShapeType || (MapShapeType = {}));
export const INFINITE_ANIMATION_ITERATIONS = 'infinite';
export let AnimationDirection;
(function (AnimationDirection) {
  AnimationDirection["NORMAL"] = "nomal";
  AnimationDirection["REVERSE"] = "reverse";
  AnimationDirection["ALTERNATE"] = "alternate";
  AnimationDirection["ALTERNATE_REVERSE"] = "alternate-reverse";
})(AnimationDirection || (AnimationDirection = {}));
